plugins/volumeFilter.go

```diff
@@ -103,8 +106,25 @@ var _ SubmitFilter = &volumeFilter{}
 
 // Validate ensures validity
 func (c *VolumeFilterConfig) Validate() error {
-	if c.isEmpty() {
-		return fmt.Errorf("the volumeFilterConfig was empty")
+	if c.SellBaseAssetCapInBaseUnits != nil && c.SellBaseAssetCapInQuoteUnits != nil {
+		return fmt.Errorf("invalid asset caps: only one asset cap can be non-nil, but both are non-nil")
+	}
+
+	if c.SellBaseAssetCapInBaseUnits == nil && c.SellBaseAssetCapInQuoteUnits == nil {
+		return fmt.Errorf("invalid asset caps: only one asset cap can be non-nil, but both are nil")
+	}
+
+	// return the original error, as it is already well-formatted
+	if _, e := parseVolumeFilterMode(string(c.mode)); e != nil {
+		return e
```

Comment:
we should always wrap the error as a habit as it helps trace the error path. long term we will need to find a less verbose solution, but this has been working well so far.